File: doc-examples/nitrite-example-java/src/main/java/example/PersonRepositoryExample.java
package example;

import io.micronaut.data.model.Page;
import io.micronaut.data.model.Pageable;
import io.micronaut.data.model.Sort;
import io.micronaut.data.repository.jpa.criteria.PredicateSpecification;
import jakarta.inject.Inject;

import java.util.Arrays;
import java.util.List;

import static example.PersonRepository.Specifications.ageIsLessThan;
import static example.PersonRepository.Specifications.nameEquals;
import static example.PersonRepository.Specifications.setNewName;
import static io.micronaut.data.repository.jpa.criteria.PredicateSpecification.not;
import static io.micronaut.data.repository.jpa.criteria.PredicateSpecification.where;

/**
 * This class exists to provide documentation snippets (it is not executed as a test).
 */
final class PersonRepositoryExample {

    @Inject PersonRepository personRepository;

    void query() {
        // tag::find[]
        Person denis = personRepository.findOne(nameEquals("Denis")).orElse(null);

        long countAgeLess30 = personRepository.count(ageIsLessThan(30));

        long countAgeLess20 = personRepository.count(ageIsLessThan(20));

        long countAgeLess30NotDenis = personRepository.count(ageIsLessThan(30).and(not(nameEquals("Denis"))));

        List<Person> people = personRepository.findAll(where(nameEquals("Denis").or(nameEquals("Josh"))));
        // end::find[]
    }

    // tag::sorting-pagination-usage[]
    void sortingAndPagination() {
        Pageable pageable = Pageable.from(0, 10, Sort.of(Sort.Order.asc("name")));
        Page<Person> page = personRepository.findAll(ageIsLessThan(30), pageable);
    }
    // end::sorting-pagination-usage[]

    void update() {
        // tag::update[]
        long recordsUpdated = personRepository.updateAll(setNewName("Steven").where(nameEquals("Denis")));
        // end::update[]
    }

    void delete() {
        // tag::delete[]
        long recordsDeleted = personRepository.deleteAll(where(nameEquals("Denis")));
        // end::delete[]
    }
}

File: doc-examples/nitrite-example-kotlin/src/main/kotlin/example/PersonRepositoryExample.kt
package example

import io.micronaut.data.model.Pageable
import io.micronaut.data.model.Sort
import io.micronaut.data.repository.jpa.criteria.PredicateSpecification.not
import io.micronaut.data.repository.jpa.criteria.PredicateSpecification.where
import jakarta.inject.Inject

/**
 * This class exists to provide documentation snippets (it is not executed as a test).
 */
class PersonRepositoryExample {

    @Inject lateinit var personRepository: PersonRepository

    fun query() {
        // tag::find[]
        val denis = personRepository.findOne(PersonRepository.Specifications.nameEquals("Denis")).orElse(null)
        val countAgeLess30 = personRepository.count(PersonRepository.Specifications.ageIsLessThan(30))
        val countAgeLess20 = personRepository.count(PersonRepository.Specifications.ageIsLessThan(20))
        val countAgeLess30NotDenis =
            personRepository.count(PersonRepository.Specifications.ageIsLessThan(30).and(not(PersonRepository.Specifications.nameEquals("Denis"))))
        val people = personRepository.findAll(where(PersonRepository.Specifications.nameEquals("Denis").or(PersonRepository.Specifications.nameEquals("Josh"))))
        // end::find[]
    }

    // tag::sorting-pagination-usage[]
    fun sortingAndPagination() {
        val pageable = Pageable.from(0, 10, Sort.of(Sort.Order.asc("name")))
        val page = personRepository.findAll(PersonRepository.Specifications.ageIsLessThan(30), pageable)
    }
    // end::sorting-pagination-usage[]

    fun update() {
        // tag::update[]
        val recordsUpdated = personRepository.updateAll(PersonRepository.Specifications.setNewName("Steven").where(PersonRepository.Specifications.nameEquals("Denis")))
        // end::update[]
    }

    fun delete() {
        // tag::delete[]
        val recordsDeleted = personRepository.deleteAll(where(PersonRepository.Specifications.nameEquals("Denis")))
        // end::delete[]
    }
}
